contracts/instruments/RibbonThetaVault.sol

```diff
@@ -419,6 +465,14 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
         currentOption = newOption;
         nextOption = address(0);
 
+        if (isYearnWrapped) {
+            uint256 amountToWrap = IERC20(asset).balanceOf(address(this));
+            // prevent spending old and new allowance through tricky tx ordering
+            collateralToken.approve(address(collateralToken), 0);
+            collateralToken.approve(address(collateralToken), amountToWrap);
```

Comment:
Would be better for gas if we avoided doing `approve(0)`, since the doing an `SSTORE` on a storage location that is zero is [20k gas instead of 5k gas](https://github.com/djrtwo/evm-opcode-gas-costs/blob/master/opcode-gas-costs_EIP-150_revision-1e18248_2017-04-12.csv#L50) for a non-zero location.